admiral/pkg/controller/admiral/rollouts.go

```diff
@@ -0,0 +1,229 @@
+	Updated(obj *argo.Rollout)
+	Deleted(obj *argo.Rollout)
+}
+
+type RolloutsEntry struct {
+	Identity string
+	Rollout *argo.Rollout
+}
+
+type RolloutClusterEntry struct {
+	Identity    string
+	Rollouts map[string][]*argo.Rollout
+}
+
+type RolloutController struct {
+	K8sClient         kubernetes.Interface
+	RolloutClient     argoprojv1alpha1.ArgoprojV1alpha1Interface
+	RolloutHandler RolloutHandler
+	informer  cache.SharedIndexInformer
+	ctl      *Controller
+	clusterName string
+	Cache    *rolloutCache
+	labelSet *common.LabelSet
+}
+
+
+type rolloutCache struct {
+	//map of dependencies key=identity value array of onboarded identities
+	cache map[string]*RolloutClusterEntry
+	mutex *sync.Mutex
+}
+
+func (p *rolloutCache) Put(rolloutEntry *RolloutClusterEntry) {
+	defer p.mutex.Unlock()
+	p.mutex.Lock()
+
+	p.cache[rolloutEntry.Identity] = rolloutEntry
+}
+
+func (p *rolloutCache) getKey(rollout *argo.Rollout) string {
+	return common.GetRolloutGlobalIdentifier(rollout)
+}
+
+func (p *rolloutCache) Get(key string) *RolloutClusterEntry {
+	return p.cache[key]
+}
+
+func (p *rolloutCache) Delete(pod *RolloutClusterEntry) {
+	defer p.mutex.Unlock()
+	p.mutex.Lock()
+	delete(p.cache, pod.Identity)
+}
+
+func (p *rolloutCache) AppendRolloutToCluster(key string, rollout *argo.Rollout) {
+	defer p.mutex.Unlock()
+	p.mutex.Lock()
+
+	v := p.Get(key)
+
+	if v == nil {
+		v = &RolloutClusterEntry{
+			Identity:    key,
+			Rollouts: make(map[string][]*argo.Rollout),
+		}
+		p.cache[v.Identity] = v
+	}
+	env := common.GetEnvForRollout(rollout)
+	envRollouts := v.Rollouts[env]
+
+	if envRollouts == nil {
+		envRollouts = make([]*argo.Rollout, 0)
+	}
+
+	envRollouts = append(envRollouts, rollout)
+
+	v.Rollouts[env] = envRollouts
+
+}
+
+func (d *RolloutController) GetRollouts() ([]*argo.Rollout, error) {
+
+	ns := d.K8sClient.CoreV1().Namespaces()
+
+	namespaceSidecarInjectionLabelFilter := d.labelSet.NamespaceSidecarInjectionLabel+"="+d.labelSet.NamespaceSidecarInjectionLabelValue
+	istioEnabledNs, err := ns.List(meta_v1.ListOptions{LabelSelector: namespaceSidecarInjectionLabelFilter})
+
+	if err != nil {
+		return nil, fmt.Errorf("error getting istio labled namespaces: %v", err)
+	}
+
+	var res []*argo.Rollout
+
+	for _, v := range istioEnabledNs.Items {
+
+		rollouts := d.RolloutClient.Rollouts(v.Name)
+		rolloutsList, err := rollouts.List(meta_v1.ListOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("error listing rollouts: %v", err)
+		}
+		var admiralRollouts []argo.Rollout
+		for _, rollout := range rolloutsList.Items {
+			if !d.shouldIgnoreBasedOnLabelsForRollout(&rollout) {
+				admiralRollouts = append(admiralRollouts, rollout)
+			}
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("error getting istio labled namespaces: %v", err)
+		}
+
+		for _, pi := range admiralRollouts {
+			res = append(res, &pi)
+		}
+	}
+
+	return res, nil
+}
+
+func (d *RolloutController) shouldIgnoreBasedOnLabelsForRollout(rollout *argo.Rollout) bool {
```

Comment:
We will only use an annotation going forward, please take a look at this PR: https://github.com/istio-ecosystem/admiral/pull/93 and make changes to allow annotation based ignore.

Reply:
Using latest deployment logic